numpy/random/mtrand/mtrand.pyx

```diff
@@ -4828,6 +4828,7 @@ cdef class RandomState:
         cdef npy_intp i, j
         for i in reversed(range(1, n)):
             j = rk_interval(i, self.internal_state)
+            if i == j : continue # i == j is not needed and memcpy is undefined.
```

Comment:
worse it would mean a datapoint cannot stay in the same position after a shuffle which biases the result